src/lib/y2network/dns.rb

```diff
@@ -21,19 +21,19 @@ module Y2Network
   # DNS configuration (hostname, nameservers, etc.).
   class DNS
     # @return [String] Hostname (local part)
-    attr_reader :hostname
+    attr_accessor :hostname
```

Comment:
I don't think that hostname belongs here. I know that in old module it was in DNS module, however this hostname (/etc/hostname) is not much related to networking and DNS.

Reply:
We already had a discussion about that and @teclator and I did not find a better place (perhaps directly under `Y2Network::Config`), so we decided to keep it here meanwhile. But I do not have a strong opinion about moving it to a different place.

Reply:
@mchf If I'm not wrong it was proposed to remove the handling of the `/etc/hostname` but not sure if it was really discussed or not and the conclusions as at least that option should be moved to other component.

So maybe the tab should just take care of DNS, and the `dhcp_hostname` option should be moved elsewhere (maybe in the General tab with the **DHCLIENT_SET_DEFAULT_ROUTE**)
I'f

Reply:
I was not talking about UI but the class responsibilities. If the class is named DNS then it should take care of DNS and not implement everything what was in the same module in the old code.